client.go

```diff
@@ -12,11 +12,15 @@ type Client struct {
 	addr       string
 	httpclient *http.Client
 	Debug      bool
+	username   string
+	password   string
 }
 
-func NewClient(addr string) (client *Client) {
+func NewClient(addr string, username string, password string) (client *Client) {
 	client = new(Client)
 	client.addr = addr
+	client.username = username
+	client.password = password
 	client.httpclient = new(http.Client)
 	return
 }
```

Comment:
I read somewhere that doing this new(xyz) then filling out the fields is inefficient as you are updating multiple times rather than just defining once